packages/astro/src/core/add/index.ts

```diff
@@ -727,6 +739,128 @@ export async function validateIntegrations(integrations: string[]): Promise<Inte
 	}
 }
 
+async function updateTSConfig(
```

Comment:
This function seems really useful, it's a shame for it to be in the add code. What do you think about:

1. Moving the logic into src/core/config/
2. Splitting it up so that there's a side-effect free version that takes a tsconfig and mutates it.
3. Adding unit tests for the side-effect version to test/units/config/ (which already contains some for other functionality in the config module)

I know this might be a bit of a refactor but I think it would be worth it to consolidate config logic into that module and have everything as testable as possible.

Reply:
That's definitely much neater, I can do that for sure!